pkg/overlay/cache.go

```diff
@@ -220,3 +231,47 @@ func randomID() ([]byte, error) {
 	_, err := rand.Read(result)
 	return result, err
 }
+
+// getNodeRep gets a node's stats from statdb and returns a node with reputation attached
+func (o *Cache) getNodeRep(ctx context.Context, n *pb.Node) (*pb.Node, error) {
+	stats, err := o.statdb.CreateEntryIfNotExists(ctx, node.IDFromString(n.Id).Bytes())
+	if err != nil {
+		return nil, err
+	}
+
+	n.Reputation = &pb.NodeRep{
+		UptimeRatio:       stats.UptimeRatio,
+		AuditSuccessRatio: stats.AuditSuccessRatio,
+		AuditCount:        stats.AuditCount,
+	}
+
+	return n, err
+}
+
+// pingNode pings a node, updates its uptime stats in statdb,
+// and returns the node with reputation attached
+func (o *Cache) pingNode(ctx context.Context, n *pb.Node) (*pb.Node, error) {
```

Comment:
nit - I feel like this could also be named `getReputation` or something. The comment is definitely helpful but I feel like it could still potentially confusing to have both `DHT.Ping` and `pingNode`.

Reply:
The problem is I have a function above `getNodeRep` already gets and returns node reputation, but without pinging. We could potentially merge the two into the same function, as long as using `DHT.Ping` in the logic that currently calls `getNodeRep` is not problematic